From 16217ab3c802830368a60b9c92f47333f20d1714 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Fri, 19 Oct 2018 11:58:39 +0200
Subject: [PATCH] Merge android-9.0.0_r12

commit c887cc301f63bc1a68348e660d0b56f4e8cc3266
Merge: 77867e0f1 a66fb8f32
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Fri Jul 20 23:48:16 2018 +0000

    Snap for 4905103 from a66fb8f32f5a956f143c8719b00076144835aa5f to pi-dr1-release

    Change-Id: If3b76406ea1654e453edac8bb96a8cdf5c0b7322

commit a66fb8f32f5a956f143c8719b00076144835aa5f
Merge: bd9634bab b59ccfdd2
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Fri Jul 20 14:41:03 2018 +0000

    Merge "111614464: Update the api-generator script" into pi-dev

commit bd9634bab54634a4384e9b91cb786db461a2f8e9
Merge: f29bc6c5d c1885b13d
Author: TreeHugger Robot <treehugger-gerrit@google.com>
Date:   Thu Jul 19 21:43:41 2018 +0000

    Merge "Update SDK tree after androidx.renderscript file name change." into pi-dev

commit b59ccfdd284810c8f24b6938361ce365bb7be280
Author: Tor Norbye <tnorbye@google.com>
Date:   Thu Jul 19 13:40:39 2018 -0700

    111614464: Update the api-generator script

    This CL updates the api generator to include the
    fix from I6f2b3bf950fbd79f81dcd0c484be400044f2c0dd
    which skips package protected members.

    Test: Lint
    Bug: 111614464

    Change-Id: Ib6a50a6ecc871c01895b4c903b4ff3be1cc917c4

commit c1885b13d86b096f60e3e4cd6d4e0df819fbe915
Author: Miao Wang <miaowang@google.com>
Date:   Fri Jun 1 11:51:40 2018 -0700

    Update SDK tree after androidx.renderscript file name change.

      - androidx.renderscript prebuilt is named as androidx-rs.
      - android.support.v8.renderscript prebuilt is still renderscript-v8.
      - Note, this ONLY impact SDK and not device images.

    Bug: 74398526
    Test: mm
    Change-Id: Ibdd547d4ce5ccb02394c2ce708561b1f13c8e16a
    (cherry picked from commit ff268369ee9d31517779f8269a7727be62db4d2f)

commit 77867e0f1adf6e3dbcf1a3b7b5a61e8388e69f67
Merge: 48bbd909c f29bc6c5d
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jul 8 03:14:01 2018 +0000

    Snap for 4880674 from f29bc6c5da24ca0c5498ac2c99b2e87f7c57c329 to pi-dr1-release

    Change-Id: I72d9ce6bd3a381af1f0114e00a0b9ca925538d30

commit f29bc6c5da24ca0c5498ac2c99b2e87f7c57c329
Author: bohu <bohu@google.com>
Date:   Fri Jul 6 08:59:15 2018 -0700

    emulator: bump sdk image revision to 4

    For DP5 release of emulator system images

    BUG: 111204833

    this cl does not impact real devices

    Change-Id: I32b0421dbadbf9dade9df33396bf041092e7cfd6

commit 48bbd909cb77aed28bef53d8488401f4326c25f5
Merge: fdb471725 41e49c338
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Jul 4 03:02:45 2018 +0000

    Snap for 4876468 from 41e49c338106534443c0ae2dfe2c75c4491d1b81 to pi-dr1-release

    Change-Id: Ief3da342e53f62238a729cde836d8480f3ae9bc1

commit 41e49c338106534443c0ae2dfe2c75c4491d1b81
Author: Colin Cross <ccross@android.com>
Date:   Mon Jul 2 11:29:03 2018 -0700

    Pass compiled metalava stub classes through metalava again

    Use metalava --rewrite-annotations to pass the compiled sdk stub
    class files through metalava again to make the annotations package
    private.

    Bug: 110532131
    Test: m out/target/common/obj/JAVA_LIBRARIES/metalava_android_stubs_current_intermediates/classes.jar
    Change-Id: I124e8d89075ec682e9fc3602ef232492bed849aa
    Merged-In: I5180bc18979415cec6255beb3fe2c48ec298d3ae

commit fdb47172587dea759573e7db6ae9f6054ce300d4
Merge: 22520ddeb 62fff1d2d
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jun 24 03:00:33 2018 +0000

    Snap for 4859140 from 62fff1d2d7677ff571e07fff9043c63fc5d30bc1 to pi-dr1-release

    Change-Id: I05b2c2d4a6e1882199bf4daba43f97f6f2d0defc

commit 62fff1d2d7677ff571e07fff9043c63fc5d30bc1
Author: Weilun Du <wdu@google.com>
Date:   Thu Jun 21 11:40:51 2018 -0700

    Copy dummy userdata.img to sdk system images

    Avd manager in Android Studio requires userdata.img in the downloaded
    system image. As we no longer inlcude the real userdata.img, we copy the
    dummy userdata.img to pass the check.
    This cl does not impact real devices.

    BUG: 110492064

    Change-Id: Ia8377ce72e32c3556a5a27100085a07f43d9917c
    Signed-off-by: Weilun Du <wdu@google.com>

commit 22520ddebbdf49f24daa5aca03edf4bdc98fcb0a
Merge: f41d065e4 f13bb3dab
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jun 17 03:04:28 2018 +0000

    Snap for 4845430 from f13bb3dab6c5617ec88d6a6ba8628ce049ed693a to pi-dr1-release

    Change-Id: Ide05f0411e3eef396125b626f9e03d91f21ff6f4

commit f13bb3dab6c5617ec88d6a6ba8628ce049ed693a
Author: Tor Norbye <tnorbye@google.com>
Date:   Wed Jun 13 18:29:45 2018 -0400

    Use android.jar from Metalava.

    This CL only affects the SDK.

    It reverts commit 91967398ad528f87d520b93049692810dbf0ca14
    and then updates the packaging source for android.jar to
    the metalava jar.

    It also packages the extracted annotations, annotations.zip.

    Test: Manually built various apps with android.jar file
      as well as some binary verification with javap
    Bug: 78245848

    Change-Id: I5614a7f5af11d12c99cf0275ae0a88d97cc0bbc7
    Merged-In: I5180bc18979415cec6255beb3fe2c48ec298d3ae

commit f41d065e47fea431c877905ecc153c43d699f2b5
Merge: 3989fdac2 266ad89c1
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Jun 14 03:02:44 2018 +0000

    Snap for 4839323 from 266ad89c16f91d688f9177e9ecf50afe32837717 to pi-dr1-release

    Change-Id: I2d048c41f2e7126dad3c8c85a8f7890cfccbc8b9

commit 266ad89c16f91d688f9177e9ecf50afe32837717
Author: bohu <bohu@google.com>
Date:   Tue Jun 12 12:05:03 2018 -0700

    emulator: use 4.4 kernel for arm images

    BUG: 109735735

    this cl does not impact real devices

    Change-Id: I56f9925bd4e0b93f2d6b71b710583e9ef298a875

commit 3989fdac23d8e14b0d5ca7e2dc6d508ace9648e6
Merge: 6cede1281 91967398a
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jun 12 03:05:20 2018 +0000

    Snap for 4834273 from 91967398ad528f87d520b93049692810dbf0ca14 to pi-dr1-release

    Change-Id: Ic5f7d99dea6dfe854b089cf797dd5e034fb22817

commit 6cede128180557b4d574c3ce27f9a6bbebc8f7c3
Merge: f7fb16a7b 3a1dcff97
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jun 10 03:07:46 2018 +0000

    Snap for 4832091 from 3a1dcff97b22cc566fc33b655d595fcd61dbb42e to pi-dr1-release

    Change-Id: I96c5603694b1aebe6c5bce3dd01109cb7e4e0a26

commit 91967398ad528f87d520b93049692810dbf0ca14
Author: Nan Zhang <nanzhang@google.com>
Date:   Fri Jun 8 21:36:18 2018 -0700

    Dist-for-goal metalava-android.jar

    Test: N/A
    Bug: b/78245848
    Change-Id: I277e1fbb57a8a77ebab789cc443cbe2a88c9633f
    Merged-In: I5180bc18979415cec6255beb3fe2c48ec298d3ae

commit f7fb16a7b3a4a9bcabbf58a4a8c1dd63479c258c
Merge: 103b63425 08ea84aef
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Jun 7 03:08:10 2018 +0000

    Snap for 4826407 from 08ea84aef42426d8a1b34a3fef636035670d8176 to pi-dr1-release

    Change-Id: I221c3e8b3794e5d5144fe5a01307f611b1b11b9f

commit 103b634254ee5eee64b30aa06ead4bddb81bce22
Merge: fb6997e6a 36f6f66a0
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jun 5 03:02:45 2018 +0000

    Snap for 4820872 from 36f6f66a0e62f9bd5a667b65553f922efd5c3b7e to pi-dr1-release

    Change-Id: I031e2f065077d163fb72fca77e98966bace30902

Change-Id: Idb5874ec440e8cf4ba6e93616913a0a562e1b933
---
 build/Android.mk                                |  2 +-
 build/build_android_stubs.mk                    | 10 +++++++++-
 build/sdk-android-arm64-v8a.atree               |  2 +-
 build/sdk-android-armeabi-v7a.atree             |  2 +-
 build/sdk-android-armeabi.atree                 |  2 +-
 build/sdk.atree                                 | 12 ++++++++----
 sys-img/images_arm64-v8a_source.prop_template   |  2 +-
 sys-img/images_armeabi-v7a_source.prop_template |  2 +-
 sys-img/images_armeabi_source.prop_template     |  2 +-
 sys-img/images_x86_64_source.prop_template      |  2 +-
 sys-img/images_x86_source.prop_template         |  2 +-
 11 files changed, 26 insertions(+), 14 deletions(-)

diff --git a/build/Android.mk b/build/Android.mk
index db55681d8..4fa56f9db 100644
--- a/build/Android.mk
+++ b/build/Android.mk
@@ -150,7 +150,7 @@ endef
 
 ALL_SDK_FILES += $(HOST_OUT)/development/sdk/generated-api-versions.xml
 
-api_gen_jar := $(TOPDIR)prebuilts/tools/common/api-generator/api-generator-26.0.0.jar
+api_gen_jar := $(TOPDIR)prebuilts/tools/common/api-generator/api-generator-26.3.0.jar
 api_gen_deps := \
   $(TOPDIR)prebuilts/tools/common/m2/repository/net/sf/kxml/kxml2/2.3.0/kxml2-2.3.0.jar \
   $(TOPDIR)prebuilts/tools/common/m2/repository/org/ow2/asm/asm/5.0.4/asm-5.0.4.jar \
diff --git a/build/build_android_stubs.mk b/build/build_android_stubs.mk
index d324934e5..393b21328 100644
--- a/build/build_android_stubs.mk
+++ b/build/build_android_stubs.mk
@@ -39,9 +39,13 @@ else
 $(full_target): PRIVATE_ERRORPRONE_FLAGS :=
 endif
 
-ifneq ($(filter metalava%,$(sdk_stub_name)),)
+my_use_metalava := $(filter metalava%,$(sdk_stub_name))
+$(warning $(sdk_stub_name) $(my_use_metalava))
+
+ifdef my_use_metalava
 $(full_target): $(HOST_OUT_JAVA_LIBRARIES)/stub-annotations$(COMMON_JAVA_PACKAGE_SUFFIX)
 $(full_target): PRIVATE_METALAVA_CLASSPATH := -classpath $(HOST_OUT_JAVA_LIBRARIES)/stub-annotations$(COMMON_JAVA_PACKAGE_SUFFIX)
+$(full_target): $(HOST_OUT_JAVA_LIBRARIES)/metalava.jar
 else
 $(full_target): PRIVATE_METALAVA_CLASSPATH :=
 endif
@@ -74,6 +78,10 @@ $(full_target): $(stub_timestamp) $(framework_res_package) $(ZIPTIME)
 			-g -d $(PRIVATE_CLASS_INTERMEDIATES_DIR) \
 			\@$(PRIVATE_INTERMEDIATES_DIR)/java-source-list \
 		|| ( rm -rf $(PRIVATE_CLASS_INTERMEDIATES_DIR) ; exit 41 )
+ifdef my_use_metalava
+	$(hide) $(JAVA) -jar $(HOST_OUT_JAVA_LIBRARIES)/metalava.jar \
+	    --no-banner --rewrite-annotations $(PRIVATE_CLASS_INTERMEDIATES_DIR)
+endif
 	$(hide) if [ ! -f $(PRIVATE_FRAMEWORK_RES_PACKAGE) ]; then \
 		echo Missing file $(PRIVATE_FRAMEWORK_RES_PACKAGE); \
 		rm -rf $(PRIVATE_CLASS_INTERMEDIATES_DIR); \
diff --git a/build/sdk-android-arm64-v8a.atree b/build/sdk-android-arm64-v8a.atree
index e5e9f78ed..56fba15cd 100644
--- a/build/sdk-android-arm64-v8a.atree
+++ b/build/sdk-android-arm64-v8a.atree
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-prebuilts/qemu-kernel/${TARGET_ARCH}/3.18/kernel-qemu2 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-ranchu
+prebuilts/qemu-kernel/${TARGET_ARCH}/4.4/kernel-qemu2 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-ranchu
 
 # version files for the SDK updater, from development.git
 ${HOST_OUT}/development/sys-img-${TARGET_CPU_ABI}/images_arm64-v8a_source.properties system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/source.properties
diff --git a/build/sdk-android-armeabi-v7a.atree b/build/sdk-android-armeabi-v7a.atree
index d488c95ce..a05229c70 100644
--- a/build/sdk-android-armeabi-v7a.atree
+++ b/build/sdk-android-armeabi-v7a.atree
@@ -15,7 +15,7 @@
 #
 
 # Copy the ARMv7 specific kernel image to .../kernel-qemu
-prebuilts/qemu-kernel/arm64/3.18/kernel-qemu2 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-ranchu-64
+prebuilts/qemu-kernel/arm64/4.4/kernel-qemu2 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-ranchu-64
 
 # version files for the SDK updater, from development.git
 ${HOST_OUT}/development/sys-img-${TARGET_CPU_ABI}/images_armeabi-v7a_source.properties system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/source.properties
diff --git a/build/sdk-android-armeabi.atree b/build/sdk-android-armeabi.atree
index 90afc82eb..293ea4c5e 100644
--- a/build/sdk-android-armeabi.atree
+++ b/build/sdk-android-armeabi.atree
@@ -15,7 +15,7 @@
 #
 
 prebuilts/qemu-kernel/${TARGET_ARCH}/kernel-qemu system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-qemu
-prebuilts/qemu-kernel/arm64/3.18/kernel-qemu2 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-ranchu-64
+prebuilts/qemu-kernel/arm64/4.4/kernel-qemu2 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/kernel-ranchu-64
 
 # version files for the SDK updater, from development.git
 ${HOST_OUT}/development/sys-img-${TARGET_CPU_ABI}/images_armeabi_source.properties system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/source.properties
diff --git a/build/sdk.atree b/build/sdk.atree
index b160cbf66..a16234df3 100644
--- a/build/sdk.atree
+++ b/build/sdk.atree
@@ -64,7 +64,7 @@ external/chromium-trace/UPSTREAM_REVISION                                 platfo
 # Compatibility: moved to platform but also leave them in platform-tools for a little
 # longer until all active Gradle plugins have support for looking in both places
 development/sdk/generated-api-versions.xml    platform-tools/api/api-versions.xml
-prebuilts/sdk/sdk-annotations/annotations.zip   platform-tools/api/annotations.zip
+${OUT_DIR}/target/common/obj/PACKAGING/metalava-api-stubs_annotations.zip   platform-tools/api/annotations.zip
 
 ##############################################################################
 # Build Tools Component
@@ -91,7 +91,8 @@ frameworks/rs/script_api/include              build-tools/${PLATFORM_NAME}/rende
 external/clang/lib/Headers                    build-tools/${PLATFORM_NAME}/renderscript/clang-include
 external/clang/LICENSE.TXT                    build-tools/${PLATFORM_NAME}/renderscript/clang-include/LICENSE.TXT
 
-prebuilts/sdk/renderscript/lib/javalib.jar            build-tools/${PLATFORM_NAME}/renderscript/lib/renderscript-v8.jar
+prebuilts/sdk/renderscript/lib/javalib.jar            build-tools/${PLATFORM_NAME}/renderscript/lib/androidx-rs.jar
+prebuilts/sdk/renderscript/lib/javalib_legacy.jar     build-tools/${PLATFORM_NAME}/renderscript/lib/renderscript-v8.jar
 
 prebuilts/sdk/renderscript/lib/arm/libc.so            build-tools/${PLATFORM_NAME}/renderscript/lib/intermediates/armeabi-v7a/libc.so
 prebuilts/sdk/renderscript/lib/arm/libm.so            build-tools/${PLATFORM_NAME}/renderscript/lib/intermediates/armeabi-v7a/libm.so
@@ -157,8 +158,9 @@ ${HOST_OUT}/development/sdk/platform_source.properties
 sdk/sdk-build.prop                                                                            platforms/${PLATFORM_NAME}/build.prop
 
 # Main Public API jar
-${OUT_DIR}/target/common/obj/PACKAGING/android_jar_intermediates/android.jar                  platforms/${PLATFORM_NAME}/android.jar
-${OUT_DIR}/target/common/obj/PACKAGING/android_jar_intermediates/android-stubs-src.jar        platforms/${PLATFORM_NAME}/android-stubs-src.jar
+${OUT_DIR}/target/common/obj/PACKAGING/metalava_android_jar_intermediates/metalava-android.jar            platforms/${PLATFORM_NAME}/android.jar
+${OUT_DIR}/target/common/obj/PACKAGING/metalava_android_jar_intermediates/metalava-android-stubs-src.jar  platforms/${PLATFORM_NAME}/android-stubs-src.jar
+
 # optional API files.
 development/build/optional.json                                                               platforms/${PLATFORM_NAME}/optional/optional.json
 ${OUT_DIR}/target/common/obj/JAVA_LIBRARIES/org.apache.http.legacy_intermediates/javalib.jar  platforms/${PLATFORM_NAME}/optional/org.apache.http.legacy.jar
@@ -204,6 +206,7 @@ development/sdk/generated-api-versions.xml  platforms/${PLATFORM_NAME}/data/api-
 
 # API annotations database for lint
 prebuilts/sdk/sdk-annotations/annotations.zip platforms/${PLATFORM_NAME}/data/annotations.zip
+${OUT_DIR}/target/common/obj/PACKAGING/metalava-api-stubs_annotations.zip platforms/${PLATFORM_NAME}/data/annotations.zip
 
 # Eclipse Editors support
 framework/layoutlib-legacy.jar            platforms/${PLATFORM_NAME}/data/layoutlib.jar
@@ -235,6 +238,7 @@ development/sdk/sdk_files_NOTICE.txt      platforms/${PLATFORM_NAME}/skins/NOTIC
 system-qemu.img                                 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/system.img
 vendor-qemu.img                                 system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/vendor.img
 ramdisk.img                                system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/ramdisk.img
+device/generic/goldfish/data/etc/userdata.img    system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/userdata.img
 data/misc                                        system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/data/misc
 system/build.prop                          system-images/${PLATFORM_NAME}/${TARGET_CPU_ABI}/build.prop
 
diff --git a/sys-img/images_arm64-v8a_source.prop_template b/sys-img/images_arm64-v8a_source.prop_template
index 4cd941eb2..c2c385180 100644
--- a/sys-img/images_arm64-v8a_source.prop_template
+++ b/sys-img/images_arm64-v8a_source.prop_template
@@ -1,6 +1,6 @@
 Pkg.Desc=Android SDK System Image
 Pkg.UserSrc=false
-Pkg.Revision=3
+Pkg.Revision=4
 AndroidVersion.ApiLevel=${PLATFORM_SDK_VERSION}
 AndroidVersion.CodeName=${PLATFORM_VERSION_CODENAME}
 SystemImage.Abi=arm64-v8a
diff --git a/sys-img/images_armeabi-v7a_source.prop_template b/sys-img/images_armeabi-v7a_source.prop_template
index 22666796e..6322547c3 100644
--- a/sys-img/images_armeabi-v7a_source.prop_template
+++ b/sys-img/images_armeabi-v7a_source.prop_template
@@ -1,6 +1,6 @@
 Pkg.Desc=Android SDK System Image
 Pkg.UserSrc=false
-Pkg.Revision=3
+Pkg.Revision=4
 AndroidVersion.ApiLevel=${PLATFORM_SDK_VERSION}
 AndroidVersion.CodeName=${PLATFORM_VERSION_CODENAME}
 SystemImage.Abi=armeabi-v7a
diff --git a/sys-img/images_armeabi_source.prop_template b/sys-img/images_armeabi_source.prop_template
index c111286d8..ebfec294c 100644
--- a/sys-img/images_armeabi_source.prop_template
+++ b/sys-img/images_armeabi_source.prop_template
@@ -1,6 +1,6 @@
 Pkg.Desc=Android SDK System Image
 Pkg.UserSrc=false
-Pkg.Revision=3
+Pkg.Revision=4
 AndroidVersion.ApiLevel=${PLATFORM_SDK_VERSION}
 AndroidVersion.CodeName=${PLATFORM_VERSION_CODENAME}
 SystemImage.Abi=armeabi
diff --git a/sys-img/images_x86_64_source.prop_template b/sys-img/images_x86_64_source.prop_template
index 9574d5de6..7f7af450c 100644
--- a/sys-img/images_x86_64_source.prop_template
+++ b/sys-img/images_x86_64_source.prop_template
@@ -1,6 +1,6 @@
 Pkg.Desc=Android SDK System Image
 Pkg.UserSrc=false
-Pkg.Revision=3
+Pkg.Revision=4
 AndroidVersion.ApiLevel=${PLATFORM_SDK_VERSION}
 AndroidVersion.CodeName=${PLATFORM_VERSION_CODENAME}
 SystemImage.Abi=x86_64
diff --git a/sys-img/images_x86_source.prop_template b/sys-img/images_x86_source.prop_template
index 6f701a2b4..ef470a024 100644
--- a/sys-img/images_x86_source.prop_template
+++ b/sys-img/images_x86_source.prop_template
@@ -1,6 +1,6 @@
 Pkg.Desc=Android SDK System Image
 Pkg.UserSrc=false
-Pkg.Revision=3
+Pkg.Revision=4
 AndroidVersion.ApiLevel=${PLATFORM_SDK_VERSION}
 AndroidVersion.CodeName=${PLATFORM_VERSION_CODENAME}
 SystemImage.Abi=x86
-- 
2.17.1

